test/rules/unnecessary_override.dart

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N unnecessary_override`
+
+class A {
+  void foo() {}
+
+  void bar() {}
+
+  int getA(Iterable a) => 0;
+
+  int getB(Iterable a) => 0;
+
+  int getC(Iterable a) => 0;
+}
+
+class B extends A {
+  @override
+  void foo() { // LINT
+    super.foo();
+  }
+
+  @override
+  void bar() { // OK
+    bar();
+  }
+
+  @override
+  int getA(Iterable a) => super.getA(a); // LINT
+
+  @override
+  int getB(Iterable a) { // LINT
+    return super.getB(a);
+  }
+
+  @override
+  int getC(Iterable a) { // OK
+    print("something");
+    return super.getC(a);
+  }
+}
```

Comment:
This needs more tests. What about overriding a method with more than one parameter, both with and without changing the order of the arguments?
```
int max(int a, int b) => super.max(a, b); // LINT
int max(int a, int b) => super.max(b, a); // OK
```
Or a method with positional optional parameters, or named parameters (where the order of the named parameters can be changed, both with and without making the override ok)?
```
int m({int a, int b}) => super.m(a: a, b: b); // LINT
int m({int a, int b}) => super.m(b: b, a: a); // LINT
int m({int a, int b}) => super.m(b: a, a: b); // OK
```
Or an operator:
```
int operator +(int other) => super + other; // LINT
```